Firestore/core/src/firebase/firestore/util/async_queue.cc

```diff
@@ -72,9 +72,7 @@ void AsyncQueue::Enqueue(const Operation& operation) {
 void AsyncQueue::EnqueueAndInitiateShutdown(const Operation& operation) {
   std::lock_guard<std::mutex> lock{shut_down_mutex_};
   VerifySequentialOrder();
-  if (is_shutting_down_) {
-    return;
-  }
```

Comment:
Can you provide more context for this change?

Reply:
See this commit comment: https://github.com/firebase/firebase-ios-sdk/pull/4795/commits/a8c4136d9e50314730e3abd6fb3e971a6c47c8db

Essentially, it was previously the case that this short circuited if already shut down, but this caused the caller to have to `EnqueueEvenAfterShutdown` to ensure that the callback was still called. Removing this short circuit simplifies `FirestoreClient::TerminateAsync`.